// googleapis/build/src/apis/digitalassetlinks/v1.js
"use strict";
/**
 * Copyright 2015 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var apirequest_1 = require("../../lib/apirequest");
// TODO: We will eventually get the `any` in here cleared out, but in the
// interim we want to turn on no-implicit-any.
// tslint:disable: no-any
// tslint:disable: class-name
// tslint:disable: variable-name
// tslint:disable: jsdoc-format
/**
 * Digital Asset Links API
 *
 * API for discovering relationships between online assets such as web sites or
 * mobile apps.
 *
 * @example
 * const google = require('googleapis');
 * const digitalassetlinks = google.digitalassetlinks('v1');
 *
 * @namespace digitalassetlinks
 * @type {Function}
 * @version v1
 * @variation v1
 * @param {object=} options Options for Digitalassetlinks
 */
var Digitalassetlinks = /** @class */ (function () {
    function Digitalassetlinks(options, google) {
        this.root = this;
        this._options = options || {};
        this.google = google;
        this.getRoot.bind(this);
        this.assetlinks = new Resource$Assetlinks(this);
        this.statements = new Resource$Statements(this);
    }
    Digitalassetlinks.prototype.getRoot = function () {
        return this.root;
    };
    return Digitalassetlinks;
}());
exports.Digitalassetlinks = Digitalassetlinks;
var Resource$Assetlinks = /** @class */ (function () {
    function Resource$Assetlinks(root) {
        this.root = root;
        this.getRoot.bind(this);
    }
    Resource$Assetlinks.prototype.getRoot = function () {
        return this.root;
    };
    Resource$Assetlinks.prototype.check = function (params, options, callback) {
        if (typeof options === 'function') {
            callback = options;
            options = {};
        }
        if (typeof params === 'function') {
            callback = params;
            params = {};
        }
        options = options || {};
        var rootUrl = options.rootUrl || 'https://digitalassetlinks.googleapis.com/';
        var parameters = {
            options: Object.assign({
                url: (rootUrl + '/v1/assetlinks:check')
                    .replace(/([^:]\/)\/+/g, '$1'),
                method: 'GET'
            }, options),
            params: params,
            requiredParams: [],
            pathParams: [],
            context: this.getRoot()
        };
        if (callback) {
            apirequest_1.createAPIRequest(parameters, callback);
        }
        else {
            return apirequest_1.createAPIRequest(parameters);
        }
    };
    return Resource$Assetlinks;
}());
exports.Resource$Assetlinks = Resource$Assetlinks;
var Resource$Statements = /** @class */ (function () {
    function Resource$Statements(root) {
        this.root = root;
        this.getRoot.bind(this);
    }
    Resource$Statements.prototype.getRoot = function () {
        return this.root;
    };
    Resource$Statements.prototype.list = function (params, options, callback) {
        if (typeof options === 'function') {
            callback = options;
            options = {};
        }
        if (typeof params === 'function') {
            callback = params;
            params = {};
        }
        options = options || {};
        var rootUrl = options.rootUrl || 'https://digitalassetlinks.googleapis.com/';
        var parameters = {
            options: Object.assign({
                url: (rootUrl + '/v1/statements:list').replace(/([^:]\/)\/+/g, '$1'),
                method: 'GET'
            }, options),
            params: params,
            requiredParams: [],
            pathParams: [],
            context: this.getRoot()
        };
        if (callback) {
            apirequest_1.createAPIRequest(parameters, callback);
        }
        else {
            return apirequest_1.createAPIRequest(parameters);
        }
    };
    return Resource$Statements;
}());
exports.Resource$Statements = Resource$Statements;
