// googleapis/build/src/apis/pagespeedonline/v4.js
"use strict";
/**
 * Copyright 2015 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var apirequest_1 = require("../../lib/apirequest");
// TODO: We will eventually get the `any` in here cleared out, but in the
// interim we want to turn on no-implicit-any.
// tslint:disable: no-any
// tslint:disable: class-name
// tslint:disable: variable-name
// tslint:disable: jsdoc-format
/**
 * PageSpeed Insights API
 *
 * Analyzes the performance of a web page and provides tailored suggestions to
 * make that page faster.
 *
 * @example
 * const google = require('googleapis');
 * const pagespeedonline = google.pagespeedonline('v4');
 *
 * @namespace pagespeedonline
 * @type {Function}
 * @version v4
 * @variation v4
 * @param {object=} options Options for Pagespeedonline
 */
var Pagespeedonline = /** @class */ (function () {
    function Pagespeedonline(options, google) {
        this.root = this;
        this._options = options || {};
        this.google = google;
        this.getRoot.bind(this);
        this.pagespeedapi = new Resource$Pagespeedapi(this);
    }
    Pagespeedonline.prototype.getRoot = function () {
        return this.root;
    };
    return Pagespeedonline;
}());
exports.Pagespeedonline = Pagespeedonline;
var Resource$Pagespeedapi = /** @class */ (function () {
    function Resource$Pagespeedapi(root) {
        this.root = root;
        this.getRoot.bind(this);
    }
    Resource$Pagespeedapi.prototype.getRoot = function () {
        return this.root;
    };
    Resource$Pagespeedapi.prototype.runpagespeed = function (params, options, callback) {
        if (typeof options === 'function') {
            callback = options;
            options = {};
        }
        if (typeof params === 'function') {
            callback = params;
            params = {};
        }
        options = options || {};
        var rootUrl = options.rootUrl || 'https://www.googleapis.com/';
        var parameters = {
            options: Object.assign({
                url: (rootUrl + '/pagespeedonline/v4/runPagespeed')
                    .replace(/([^:]\/)\/+/g, '$1'),
                method: 'GET'
            }, options),
            params: params,
            requiredParams: ['url'],
            pathParams: [],
            context: this.getRoot()
        };
        if (callback) {
            apirequest_1.createAPIRequest(parameters, callback);
        }
        else {
            return apirequest_1.createAPIRequest(parameters);
        }
    };
    return Resource$Pagespeedapi;
}());
exports.Resource$Pagespeedapi = Resource$Pagespeedapi;
